Gems/Atom/RHI/Vulkan/Code/Source/RHI/StreamingImagePool.cpp

```diff
@@ -59,27 +87,68 @@ namespace AZ
                 }
             }
 
-            outHeapTiles = m_tileAllocator.Allocate(blockCount);
-            if (outHeapTiles.size() == 0)
+            AZStd::vector<VmaAllocation> allocations(blockCount);
+            VmaAllocationCreateInfo allocCreateInfo = GetVmaAllocationCreateInfo(RHI::HeapMemoryLevel::Device);
+
+            auto& device = static_cast<Device&>(GetDevice());
+            VkResult vkResult = vmaAllocateMemoryPages(
+                device.GetVmaAllocator(),
+                &memReq,
+                &allocCreateInfo,
+                blockCount,
+                allocations.data(),
+                nullptr);
+
+            AssertSuccess(vkResult);
+            if (vkResult == VK_SUCCESS)
             {
-                return RHI::ResultCode::OutOfMemory;
-            }
-            else
-            {
-                return RHI::ResultCode::Success;
+                AZStd::transform(
```

Comment:
This is fine but I prefer good old fashioned loops :) Maybe add in a  quick comment what you are doing here.